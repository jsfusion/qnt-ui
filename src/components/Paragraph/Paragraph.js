import React from 'react';
import { PropTypes } from 'prop-types';
import { ThemeProvider } from 'styled-components';
import cx from 'classnames';
import { appPrefix } from '../../settings';
import { StyledParagraph } from './styles/Paragraph.styles';

const Paragraph = React.forwardRef(
  (
    {
      children,
      className: customClassName,
      id,
      isBold = false,
      isItalic = false,
      isAssistive = false,
      isJustify = false,
      ...rest
    },
    ref,
  ) => {
    const className = cx({
      [`${appPrefix}-paragraph`]: true,
      [`${appPrefix}-paragraph--bold`]: isBold,
      [`${appPrefix}-paragraph--italic`]: isItalic,
      [`${appPrefix}-paragraph--assistive`]: isAssistive,
      [`${appPrefix}-paragraph--justify`]: isJustify,
      [customClassName]: !!customClassName,
    });

    return (
      <ThemeProvider theme={{}}>
        <StyledParagraph id={id} ref={ref} className={className} {...rest}>
          {children}
        </StyledParagraph>
      </ThemeProvider>
    );
  },
);

Paragraph.propTypes = {
  id: PropTypes.string,
  isBold: PropTypes.bool,
  isItalic: PropTypes.bool,
  isAssistive: PropTypes.bool,
  isJustify: PropTypes.bool,
};

export default Paragraph;
