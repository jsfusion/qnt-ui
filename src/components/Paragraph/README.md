# Paragraph

Some description...

## Usage

```js
import { Paragraph } from '@qnt/ui';
```

```jsx
<Paragraph></Paragraph>
```

## Prop Types

**id** `string`<br />
To identify the element

---
