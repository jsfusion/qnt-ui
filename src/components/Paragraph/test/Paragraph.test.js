import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Paragraph from '..';

configure({ adapter: new Adapter() });

let wrapper;
const props = {
  id: 'test_paragraph_id',
  isBold: true,
  isItalic: true,
  isJustify: true,
};

describe('***--- Paragraph Component ---***', () => {
  beforeAll(() => {
    wrapper = shallow(<Paragraph {...props}>Paragraph text</Paragraph>);
  });

  it('To match snapshot of Paragraph component', async (done) => {
    expect(wrapper).toMatchSnapshot();
    done();
  });
});