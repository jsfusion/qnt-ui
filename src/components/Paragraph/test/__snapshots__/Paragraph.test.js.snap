// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`***--- Paragraph Component ---*** To match snapshot of Paragraph component 1`] = `ShallowWrapper {}`;
